### PR TITLE
feat: Enable custom attributes in the suggestion for variables.

### PR DESCRIPTION
**Pull request by muhsin-k**
Fixes https://linear.app/chatwoot/issue/CW-2880/frontend-enable-custom-attributes-in-the-suggestion-for-variables

**Preview**

**Variables**

Screenshot 2023-12-14 at 7 32 03 PM

**Canned responses**

Screenshot 2023-12-14 at 7 33 09 PM

### CODE REVIEW

**Comment by pranavrajs** [comment]
@muhsin-k @absurdiya We need a new design for this one. The text is not readable. 

**Before:** 
Screenshot 2023-12-10 at 10 15 25 PM 



**Suggestion:** 
Frame 1 (1) |

**Review by pranavrajs — CHANGES_REQUESTED** [review]
Please see the comment on the design change.

**Comment by absurdiya** [comment]
> Please see the comment on the design change.
Screenshot 2023-12-11 at 12 28 18 PM
Would suggest keeping the hover state like this (bg: blue-2 from Radix (#F5FAFF) and text as primary color), as that is the hover state we are currently using in lists.

**Review by scmmishra — COMMENTED** [review]
@muhsin-k the code seems fine, lets work on the design suggestions from @pranavrajs and @absurdiya

**Comment by muhsin-k** [comment]
@pranavrajs I have updated the UI.

**Variables**

Screenshot 2023-12-14 at 7 32 03 PM

**Canned responses**

Screenshot 2023-12-14 at 7 33 09 PM

**Comment by scmmishra** [comment]
@pranavrajs @absurdiya please take a look whenever you can

**Comment by pranavrajs** [comment]
@muhsin-k There is a bug. Please see the screenshot. The value is defined. However, it returns this error.
Screenshot 2023-12-18 at 4 49 45 PM

**Comment by muhsin-k** [comment]
> @muhsin-k There is a bug. Please see the screenshot. The value is defined. However, it returns this error. Screenshot 2023-12-18 at 4 49 45 PM

@pranavrajs Did you run 'yarn install' after pulling the branch? The logic for this operation exists in the utils library.

File: yarn.lock [hunk 1]
```diff
@@ -3177,10 +3177,10 @@
     prosemirror-utils "^0.9.6"
     prosemirror-view "^1.17.2"
 
-"@chatwoot/utils@^0.0.16":
-  version "0.0.16"
-  resolved "https://registry.yarnpkg.com/@chatwoot/utils/-/utils-0.0.16.tgz#84a0ade7f0377a3c7a95b2e5fa3c3b3d7d045339"
-  integrity sha512-PcWEJ+LeJlDkjOOCHyi7wLaaBsPpedDh1jBblkqOYnt8PvKHSttq/Fusi6u6EcxPhIw1umgq5A/k8sWozmZ4iQ==
+"@chatwoot/utils@0.0.18":
```
Comment (iamsivin) [inline yarn.lock:3180]: @muhsin-k Do we need this version change? In pakage.json, the version is `0.0.16`.

**Review by iamsivin — APPROVED** [review]
@muhsin-k I have tested this PR locally with canned responses and variables, and it appears to be functioning properly. I have attached a Loom video below for reference.

I also have some suggestions. Please feel free to ignore them if they are not relevant.
1. When the checkbox attribute is added and its value is set to false, attempting to send a message triggers an undefined modal. Clicking 'Yes' in this modal sends a message with the value 'false'. Do we need to display the modal in this scenario?

2. Regarding the date custom attributes: the date displayed in the sidebar is formatted as '20/12/2023', but when sent, it appears in the message bubble in the format '2023-12-19T18:30:00.000Z'. Should we standardize the date format for consistency?
https://www.loom.com/share/784d0f4920db467c80e88679401ca6bf?sid=4446da15-0ce2-4f33-a2d8-0100c8974b9f

File: yarn.lock (hunk 1 above)
Comment (iamsivin) [inline yarn.lock:3180]:
```suggestion
"@chatwoot/utils@^0.0.18":
```

**Comment by muhsin-k** [comment]
>
> 1. When the checkbox attribute is added and its value is set to false, attempting to send a message triggers an undefined modal. Clicking 'Yes' in this modal sends a message with the value 'false'. Do we need to display the modal in this scenario?

I have added fix for this issue.

> 2. Regarding the date custom attributes: the date displayed in the sidebar is formatted as '20/12/2023', but when sent, it appears in the message bubble in the format '2023-12-19T18:30:00.000Z'. Should we standardize the date format for consistency?

I will handle this as a follow-up ticket.

**Pull request merged**